src/debug.cpp

```diff
@@ -54,4 +58,375 @@ void printStringStringMap(const StringStringMap &map)
     }
 }
 
+static const std::string SPACES = "  ";
+static const std::string TRUNK = "  │";
+
+struct GeneratorEquationAstTrunk
+{
+    GeneratorEquationAstTrunk *mPrev;
+    std::string mStr;
+
+    GeneratorEquationAstTrunk(GeneratorEquationAstTrunk *prev,
+                              const std::string &str);
+};
+
+GeneratorEquationAstTrunk::GeneratorEquationAstTrunk(GeneratorEquationAstTrunk *prev,
+                                                     const std::string &str)
+    : mPrev(prev)
+    , mStr(str)
+{
+}
+
+void doPrintAst(GeneratorEquationAstTrunk *trunk)
+{
+    if (trunk == nullptr) {
+        return;
+    }
+
+    doPrintAst(trunk->mPrev);
+
+    if ((trunk->mPrev != nullptr) && (trunk->mPrev->mStr == SPACES)
+        && ((trunk->mStr == SPACES) || (trunk->mStr == TRUNK))) {
+        std::cout << " ";
+    }
+
+    std::cout << trunk->mStr;
+}
+
+std::string doPrintAst(const GeneratorEquationAstPtr &ast)
```

Comment:
I don't think we need all the spacing in this function.

Reply:
Yes, I guess that this has never really been discussed as such. It's true that I like my code to be "spacious", but (most of?) the libCellML code is not so. Happy for you to edit as you see fit.

Reply:
I shall consider this conversation resolved as I am not sure there is a need to remove spaces as such to have this PR approved. However, I will definitely try to mimic more closely the style used in the rest of the library (yuck! :p :)).